classlib/src/main/java/org/teavm/classlib/java/util/random/TRandomGenerator.java

```diff
@@ -0,0 +1,298 @@
+                int res = nextInt();
+                if (res >= origin && res < bound) {
+                    return res;
+                }
+            }
+        }
+    }
+
+    long nextLong();
+
+    default long nextLong(long bound) {
+        checkBound(bound);
+        long mask = (Long.highestOneBit(bound) << 1) - 1;
+        while (true) {
+            long res = nextLong() & mask;
+            if (res < bound) {
+                return res;
+            }
+        }
+    }
+
+    default long nextLong(long origin, long bound) {
+        checkRange(origin, bound);
+        long range = bound - origin;
+        if (range > 0) {
+            return nextLong(range) + origin;
+        } else {
+            while (true) {
+                long res = nextLong();
+                if (res >= origin && res < bound) {
+                    return res;
+                }
+            }
+        }
+    }
+
+    default double nextGaussian() {
+        return pairGaussian(this)[0];
+    }
+
+    default double nextGaussian(double mean, double stddev) {
+        if (stddev < 0.0) {
+            throw new IllegalArgumentException();
+        }
+        return mean + stddev * nextGaussian();
+    }
+
+    //********************************** UTILITY *************************************************//
+
+    private static void checkStreamSize(long streamSize) {
+        if (streamSize < 0L) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    private static void checkBound(float bound) {
+        if (!(bound > 0.0 && bound < Float.POSITIVE_INFINITY)) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    private static void checkBound(double bound) {
+        if (!(bound > 0.0 && bound < Double.POSITIVE_INFINITY)) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    private static void checkBound(int bound) {
+        if (bound <= 0) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    private static void checkBound(long bound) {
+        if (bound <= 0) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    private static void checkRange(float origin, float bound) {
+        if (!(origin < bound && bound - origin < Float.POSITIVE_INFINITY)) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    private static void checkRange(double origin, double bound) {
+        if (!(origin < bound && bound - origin < Double.POSITIVE_INFINITY)) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    private static void checkRange(int origin, int bound) {
+        if (origin >= bound) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    private static void checkRange(long origin, long bound) {
+        if (origin >= bound) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    static double[] pairGaussian(TRandomGenerator rng) {
```

Comment:
I'd moved this method into separate class  somewhere in `org.teavm.classlib.impl` package.